lib/shopify_app/middleware/same_site_cookie_middleware.rb

```diff
@@ -0,0 +1,58 @@
+module ShopifyApp
+  class SameSiteCookieMiddleware
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      _status, headers, _body = @app.call(env)
+    ensure
+      user_agent = env['HTTP_USER_AGENT']
+
+      if headers && headers['Set-Cookie'] && !SameSiteCookieMiddleware.same_site_none_incompatible?(user_agent) &&
+        ShopifyApp.configuration.embedded_app == true
```

Comment:
From your PR description:

> Also tophatted with one of our first party embedded apps Exchange (though I removed the ShopifyApp.configuration.embedded_app == true check to test as Exchange runs both as an embedded app and non embedded app).

To me, this is a smell that `ShopifyApp.configuration.embedded_app == true` might not be the right configuration option to rely on. Would it make sense to add a configuration option that is on by default if the application is embedded, but can be turned on by developers even if their application is not embedded? This would also allow more configurability for embedded app authors who may want to handle this in their own app.